cfpb/ccdb5-ui
# DATAP-379 Product Filter class -> functional component, disable options when in Product Focus mode

Addresses: DATAP-1353 , DATAP-379

Convert Product filter to functional syntax

## Additions

- Jest tests
- fixed some linting issues with .substr to substring

## Removals

- snapshot test

## Changes

- moved logic to process filter options into a separate testable function

## Testing

- yarn test, tests should pass
- go to Trends
- Under subproducts, click a product filter row (Credit reporting, credit repair, services or other personal ..." to expand it
- Click "visualize sub-product and issue trends for Credit reporting, credit repair services, or other personal consumer reports"
- The filter should be selected and all other Product filters should be disabled
- You should still be able to select the Sub-products under "Credit reporting, credit repair, services or other personal"

<img width="1354" alt="Screenshot 2024-04-15 at 2 17 09 PM" src="https://github.com/cfpb/ccdb5-ui/assets/1038306/13e7ea00-061c-4bdb-8ae9-0bf9890c0849">

## Screenshots


## Notes

-

## Todos

-

## Checklist
